test/e2e/walkthrough.go

```diff
@@ -0,0 +1,209 @@
+			bindingName = "ups-binding"
+		)
+
+		//Broker and ServiceClass should become ready
+		By("Make sure the named Broker not exist before create")
+		if _, err := f.ServiceCatalogClientSet.ServicecatalogV1alpha1().Brokers().Get(brokerName, metav1.GetOptions{}); err == nil {
+			err = f.ServiceCatalogClientSet.ServicecatalogV1alpha1().Brokers().Delete(brokerName, nil)
+			Expect(err).NotTo(HaveOccurred(), "failed to delete the broker")
+
+			By("Waiting for Broker to not exist")
+			err = util.WaitForBrokerToNotExist(f.ServiceCatalogClientSet.ServicecatalogV1alpha1(), brokerName)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		By("Creating a Broker")
+		url := "http://" + upsbrokername + "." +f.Namespace.Name + ".svc.cluster.local"
+		broker := &v1alpha1.Broker{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: brokerName,
+			},
+			Spec: v1alpha1.BrokerSpec{
+				URL: url,
+			},
+		}
+		broker, err := f.ServiceCatalogClientSet.ServicecatalogV1alpha1().Brokers().Create(broker)
+		Expect(err).NotTo(HaveOccurred(), "failed to create Broker")
+
+		By("Waiting for Broker to be ready")
+		err = util.WaitForBrokerCondition(f.ServiceCatalogClientSet.ServicecatalogV1alpha1(),
+			broker.Name,
+			v1alpha1.BrokerCondition{
+				Type: v1alpha1.BrokerConditionReady,
+				Status: v1alpha1.ConditionTrue,
+			},
+		)
+		Expect(err).NotTo(HaveOccurred(), "failed to wait Broker to be ready")
+
+		By("Waiting for ServiceClass to be ready")
+		err = util.WaitForServiceClassToExist(f.ServiceCatalogClientSet.ServicecatalogV1alpha1(), serviceclassName)
+		Expect(err).NotTo(HaveOccurred(), "failed to wait serviceclass to be ready")
+
+		//Provisioning a Instance and binding to it
+		By("Creating a namespace")
+		testnamespace, err := framework.CreateKubeNamespace(testns, f.KubeClientSet)
+		Expect(err).NotTo(HaveOccurred(), "failed to create kube namespace")
+
+		By("Creating a Instance")
+		instance := &v1alpha1.Instance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: instanceName,
+				Namespace: testnamespace.Name,
+			},
+			Spec: v1alpha1.InstanceSpec{
+				ServiceClassName: serviceclassName,
+				PlanName: "default",
+			},
+		}
+		instance, err = f.ServiceCatalogClientSet.ServicecatalogV1alpha1().Instances(testnamespace.Name).Create(instance)
+		Expect(err).NotTo(HaveOccurred(), "failed to create instance")
+
+		By("Waiting for Instance to be ready")
+		err = util.WaitForInstanceCondition(f.ServiceCatalogClientSet.ServicecatalogV1alpha1(),
+			testnamespace.Name,
+			instanceName,
+			v1alpha1.InstanceCondition{
+				Type: v1alpha1.InstanceConditionReady,
+				Status: v1alpha1.ConditionTrue,
+			},
+		)
+		Expect(err).NotTo(HaveOccurred(), "failed to wait instance to be ready")
+
+		//Binding to the Instance
+		By("Creating a Binding")
+		binding := &v1alpha1.Binding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: bindingName,
+				Namespace: testnamespace.Name,
+			},
+			Spec: v1alpha1.BindingSpec{
+				InstanceRef: v1.LocalObjectReference{
```

Comment:
@duglin sorry for the response latency, see my reply at https://github.com/kubernetes-incubator/service-catalog/issues/890#issuecomment-304471395